dask/compose.py

```diff
@@ -0,0 +1,66 @@
+from uuid import uuid4
+import operator
+from functools import partial, wraps
+
+from toolz import merge
+
+from .optimize import cull
+from .context import _globals
+from . import threaded
+
+
+def get_uuid_dasks(v):
+    if isinstance(v, Value):
+        return v._uuid, v._dasks
+    else:
+        return v, []
+
+
+def applyfunc(func, *args, **kwargs):
+    func = partial(func, **kwargs)
+    uuids, dsks = zip(*map(get_uuid_dasks, args))
+    uuid = uuid4().hex
+    dsks = sum(dsks, [{uuid: (func,) + uuids}])
+    return Value(uuid, dsks)
+
+
+def daskify(func):
+    """Wraps func so that it outputs a ``Value``"""
+    @wraps(func)
+    def _dfunc(*args, **kwargs):
+        return applyfunc(func, *args, **kwargs)
+    return _dfunc
```

Comment:
> Do we want daskified functions to evaluate normally on non-values?

I don't think so, that's the whole point of doing it this way. But perhaps. This might solve the iterating through a long list of values or data problem.

> Alternatively should daskify on a value produce a Value?

That could be a nice uniting interface.